### PR TITLE
ObjectRef: fix some v3f checks

### PR DESCRIPTION
**Pull request by Zughy**
Resolves a few crashes inadvertently implemented with #10512

## To do

This PR is Ready for Review.

## How to test

Check diffs

### CODE REVIEW

File: doc/lua_api.txt [hunk 1]
```diff
@@ -6516,8 +6517,8 @@ object you are working with still exists.
     * `frame_speed` sets the animations frame speed. Default is 30.
 * `get_local_animation()`: returns idle, walk, dig, walk_while_dig tables and
   `frame_speed`.
-* `set_eye_offset(firstperson, thirdperson)`: defines offset vectors for camera
-  per player.
+* `set_eye_offset([firstperson, thirdperson])`: defines offset vectors for
+  camera per player. When a field is not specified is `{x=0, y=0, z=0}` by default.
```
Comment (sfan5) [inline doc/lua_api.txt:6521]:
```suggestion
  camera per player. When a field is not specified it is `{x=0, y=0, z=0}` by default.
```

File: doc/lua_api.txt [hunk 2]
```diff
@@ -6185,21 +6185,22 @@ object you are working with still exists.
...
 * `set_animation_frame_speed(frame_speed)`
     * `frame_speed`: number, default: `15.0`
-* `set_attach(parent, bone, position, rotation, forced_visible)`
+* `set_attach(parent, bone[, position, rotation, forced_visible])`
     * `bone`: string
-    * `position`: `{x=num, y=num, z=num}` (relative)
-    * `rotation`: `{x=num, y=num, z=num}` = Rotation on each axis, in degrees
+    * `position`: `{x=num, y=num, z=num}`, relative, default `{x=0, y=0, z=0}`
+    * `rotation`: `{x=num, y=num, z=num}` = Rotation on each axis, in degrees.
+      Default `{x=0, y=0, z=0}`.
```
Comment (sfan5) [inline doc/lua_api.txt:6192]:
```suggestion
      Default `{x=0, y=0, z=0}`
```

File: doc/lua_api.txt [hunk 3]
```diff
@@ -6185,21 +6185,22 @@ object you are working with still exists.
...
     or nil if it isn't attached.
 * `get_children()`: returns a list of ObjectRefs that are attached to the
     object.
 * `set_detach()`
-* `set_bone_position(bone, position, rotation)`
-    * `bone`: string
-    * `position`: `{x=num, y=num, z=num}` (relative)
-    * `rotation`: `{x=num, y=num, z=num}`
+* `set_bone_position([bone, position, rotation])`
+    * `bone`: string. Default is `""`, the root bone
```
Comment (rubenwardy) [inline doc/lua_api.txt:6201]: calling this with no arguments doesn't make sense to me

File: doc/lua_api.txt [hunk 4]
```diff
@@ -6516,8 +6517,8 @@ object you are working with still exists.
     * `frame_speed` sets the animations frame speed. Default is 30.
 * `get_local_animation()`: returns idle, walk, dig, walk_while_dig tables and
   `frame_speed`.
-* `set_eye_offset(firstperson, thirdperson)`: defines offset vectors for camera
-  per player.
+* `set_eye_offset([firstperson, thirdperson])`: defines offset vectors for
```
Comment (rubenwardy) [inline doc/lua_api.txt:6520]: calling this with no arguments doesn't make sense to me. I don't think these are even the correct defaults?

File: src/script/lua_api/l_object.cpp [hunk 5]
```diff
@@ -2185,7 +2182,7 @@ int ObjectRef::l_set_minimap_modes(lua_State *L)
 
 	luaL_checktype(L, 2, LUA_TTABLE);
 	std::vector<MinimapMode> modes;
-	s16 selected_mode = luaL_checkint(L, 3);
+	s16 selected_mode = readParam<s16>(L, 3);
```
Comment (rubenwardy) [inline src/script/lua_api/l_object.cpp:2185]: This should be required

**Review by rubenwardy — COMMENTED** [review]
It's unclear what crashes this fixes, but I like making some of the `set_attach` optional

File: doc/lua_api.txt (hunk 3 above)
Comment (rubenwardy) [reply doc/lua_api.txt:6201]: sfan5 claims this is desired

File: doc/lua_api.txt (hunk 4 above)
Comment (appgurueu) [reply doc/lua_api.txt:6520]: This makes sense in order to reset eye offset for example after detaching from an entity. The default values are correct.

Comment (rubenwardy) [reply doc/lua_api.txt:6520]: I would have thought that they would have the same zero position, meaning that setting the third person offset to 0,0,0 would disable it?

**Review by sfan5 — APPROVED** [review]
Looks good in the current state.

**Review by SmallJoker — APPROVED** [review]

**Pull request merged**